engine/src/main/java/org/camunda/bpm/engine/impl/cfg/ProcessEngineConfigurationImpl.java

```diff
@@ -2113,8 +2113,11 @@ protected void initMetrics() {
   }
 
   protected void initHostName() {
-    if (hostnameProvider == null) {
-      hostnameProvider = new SimpleIpBasedProvider();
+    if (hostname == null) {
```

Comment:
If I set the `hostname` via config, there will be no `hostnameProvider` in the config anymore.
The type check in `ProcessEngineImpl#L113-114` for `SimpleIpBasedProvider` will then eventually not behave as expected, will it?